source/javascripts/controllers/awards_controller.coffee

```diff
@@ -1,2 +1,17 @@
-@EGA.controller "AwardsController", ($scope, $http) ->
-  $http.get("data/awards.json").then (res) -> $scope.awards = res.data
+EGA.controller "AwardsController", ($scope, $http) ->
+  $scope.awards = []
+  $scope.categories = []
+  $scope.categoryIsShown = {}
+  $scope.years = []
+  $scope.yearIsShown = {}
+
+  $http.get("data/awards.json").then (res) ->
+    $scope.awards = res.data
+
+    categoriesFilter = wirsing.filter['switch'].on($scope.awards, 'category')
```

Comment:
Oh, I remember why I did it this way: Because I contemplated whether I should make it a factory function (like `wirsing.filter('switch')`) or just keep it as an object (a `Hash` in my Ruby mind :) ). So this thinking of it as a hash made me choose this notation without thinking about it.

I didn't make it a function because I couldn't see a real advantage of it for now.